CRM/Core/BAO/CustomGroup.php

```diff
@@ -1432,7 +1432,12 @@ public static function setDefaults(&$groupTree, &$defaults, $viewMode = FALSE, $
           }
           else {
             if ($field['data_type'] == "Float") {
-              $defaults[$elementName] = (float) $value;
+              if ($field['html_type'] == 'Text') {
+                $defaults[$elementName] = CRM_Utils_Number::formatLocaleNumeric($value);
+              }
+              else {
+                $defaults[$elementName] = (float) $value;
```

Comment:
Ah - maybe that makes more sense than what I did

Reply:
OK - so I definitely agree with this part of the PR & could quick merge this part